docs/GettingStartedDocs/OP-TEE/Debugging/QEMU.md

```diff
@@ -0,0 +1,418 @@
+debugging purposes, consider turning position-independent code generation off
+temporarily.
+
+## Loading Enclave Symbols
+
+Instructing GDB to load symbols for an enclave requires some work the first
+time.
+
+After launching QEMU, copying the sample enclave into it and launching the
+corresponding host application, OP-TEE prints lines similar to the following in
+the Secure World XTerm window:
+
+```
+D/TC:? 0 system_open_ta_binary:286 Lookup user TA ELF 126830b9-eb9f-412a-89a7-bcc8a517c12e (Secure Storage TA)
+D/TC:? 0 system_open_ta_binary:289 res=0xffff0008
+D/TC:? 0 system_open_ta_binary:286 Lookup user TA ELF 126830b9-eb9f-412a-89a7-bcc8a517c12e (REE [buffered])
+D/TC:? 0 system_open_ta_binary:289 res=0x0
+D/LD:  ldelf:150 ELF (126830b9-eb9f-412a-89a7-bcc8a517c12e) at 0x40010000
+```
+
+The fifth line indicates where in secure virtual memory OP-TEE has loaded the
+enclave. In this case it's `0x40010000`. This value should remain the same
+throughout repeated runs as well as across reboots of the emulator. Note this
+value.
+
+Due to how OP-TEE loads enclaves, you must manually line up the symbols in the
+ELF file produced for enclaves with how the code is laid out in memory:
+
+```bash
+#[ TERM 3 ]
+
+cd $HOME/openenclave_qemu
+
+./emulation/toolchains/aarch64/bin/aarch64-linux-gnu-objdump -x \
+    ./build/tests/hexdump/enc/126830b9-eb9f-412a-89a7-bcc8a517c12e.elf | less
+```
+
+In the `Sections` table, you will see output like this:
+
+```
+Sections:
+Idx Name          Size      VMA               LMA               File off  Algn
+  0 .ta_head      00000020  0000000000000000  0000000000000000  00001000  2**12
+                  CONTENTS, ALLOC, LOAD, READONLY, DATA
+  1 .text         0007d9a8  0000000000000020  0000000000000020  00001020  2**3
+                  CONTENTS, ALLOC, LOAD, READONLY, CODE
+  2 .eh_frame     0000bcd0  000000000007d9c8  000000000007d9c8  0007e9c8  2**3
+                  CONTENTS, ALLOC, LOAD, READONLY, DATA
+  3 .rodata       00009cb5  00000000000896a0  00000000000896a0  0008a6a0  2**4
+                  CONTENTS, ALLOC, LOAD, READONLY, DATA
+```
+
+Take note of the `LMA` (Load Memory Address) of the `.text` section.
+
+Add the value you noted before from OP-TEE's output to the value of `LMA` for
+the `.text` section. In this case:
+
+```
+0x40010000 + 0x20 = 0x40010020
+```
+
+Switch back to `TERM 2` where GDB is running and type:
+
+```
+<CTRL-c>
+add-symbol-file ./build/tests/hexdump/enc/126830b9-eb9f-412a-89a7-bcc8a517c12e.elf 0x40010020
+```
+
+From this point forward, even if you rebuild enclaves, the load address for the
+symbols should remain the same, so you do not need to go through of all this
+every time you want to debug. Just execute the `add-symbol-file` command in GDB.
+However, should your breakpoints stop being hit, do verify that these addresses
+have not changed, especially after a `git pull` and a rebuild.
+
+There is a known issue, either in GDB or QEMU, where once OP-TEE and Linux have
```

Comment:
Is there a github issue # that you can reference in the comment here?

Reply:
Added.